packages/react-query/src/utils/createUtilityFunctions.ts

```diff
@@ -145,5 +145,31 @@ export function createUtilityFunctions<TRouter extends AnyRouter>(
     getInfiniteQueryData: (queryKey) => {
       return queryClient.getQueryData(queryKey);
     },
+
+    setMutationDefaults: (mutationKey, options) => {
+      const path = mutationKey[0];
```

Comment:
Again, I'm not very familiar with these functions but why don't we use the full key?

Reply:
Because we need the path to effectively call the mutation function on the tRPC client. A similar thing is done in the `useMutation` definition. This is basically just the inverse formula of:

```ts
const mutationKey = [path];
```

![Screenshot 2024-07-05 alle 15 05 41](https://github.com/trpc/trpc/assets/69400730/abb52420-0fce-46df-8713-fa45f6e1a983)

Reply:
Oh right... read that one incorrectly 🫠